src/gcp/identityPlatform.ts

```diff
@@ -0,0 +1,221 @@
+      resetPasswordTemplate: EmailTemplate;
+      verifyEmailTemplate: EmailTemplate;
+      changeEmailTemplate: EmailTemplate;
+      legacyResetPasswordTemplate: EmailTemplate;
+      callbackUri: string;
+      dnsInfo: {
+        customDomain: string;
+        useCustomDomain: boolean;
+        pendingCustomDomain: string;
+        customDomainState:
+          | "VERIFICATION_STATE_UNSPECIFIED"
+          | "NOT_STARTED"
+          | "IN_PROGRESS"
+          | "FAILED"
+          | "SUCCEEDED";
+        domainVerificationRequestTime: string;
+      };
+      revertSecondFactorAdditionTemplate: EmailTemplate;
+      smtp: {
+        senderEmail: string;
+        host: string;
+        port: number;
+        username: string;
+        password: string;
+        securityMode: "SECURITY_MODE_UNSPECIFIED" | "SSL" | "START_TLS";
+      };
+    };
+    sendSms: {
+      useDeviceLocale?: boolean;
+      smsTemplate?: {
+        content?: string;
+      };
+    };
+    defaultLocale?: string;
+  };
+  quota?: {
+    signUpQuotaConfig?: {
+      quota?: string;
+      startTime?: string;
+      quotaDuration?: string;
+    };
+  };
+  monitoring?: {
+    requestLogging?: {
+      enabled?: boolean;
+    };
+  };
+  multiTenant?: {
+    allowTenants?: boolean;
+    defaultTenantLocation?: string;
+  };
+  authorizedDomains?: Array<string>;
+  subtype?: "SUBTYPE_UNSPECIFIED" | "IDENTITY_PLATFORM" | "FIREBASE_AUTH";
+  client?: {
+    apiKey?: string;
+    permissions?: {
+      disabledUserSignup?: boolean;
+      disabledUserDeletion?: boolean;
+    };
+    firebaseSubdomain?: string;
+  };
+  mfa?: {
+    state?: "STATE_UNSPECIFIED" | "DISABLED" | "ENABLED" | "MANDATORY";
+    enabledProviders?: Array<Provider>;
+  };
+  blockingFunctions?: BlockingFunctionsConfig;
+}
+
+/**
+ * Helper function to get the blocking function config from identity platform.
+ * @param project GCP project ID or number
+ * @returns the blocking functions config
+ */
+export async function getBlockingFunctionsConfig(
+  project: string
+): Promise<BlockingFunctionsConfig> {
+  const config = (await getConfig(project)) || {};
+  if (!config.blockingFunctions) {
+    config.blockingFunctions = {};
+  }
+  return config.blockingFunctions;
+}
+
+/**
+ * Gets the identity platform configuration.
+ * @param project GCP project ID or number
+ * @returns the identity platform config
+ */
+export async function getConfig(project: string): Promise<Config> {
+  const response = await adminApiClient.get<Config>(`projects/${project}/config`);
+  return response.body;
+}
+
+/**
+ * Helper function to set the blocking function config to identity platform.
+ * @param project GCP project ID or number
+ * @param blockingConfig the blocking functions configuration to update
+ * @returns the blocking functions config
+ */
+export async function setBlockingFunctionsConfig(
+  project: string,
+  blockingConfig: BlockingFunctionsConfig
+): Promise<BlockingFunctionsConfig> {
+  const config =
```

Comment:
Nit: Would this be easier if you just used `proto.fieldMasks` to create your field mask instead of setting it to "blockingFunctions"? Then you wouldn't need to merge both beforeCreate and beforeSignIn before setting it.

Reply:
So funny thing about their API... if you try and use a field mask that has the inner property that changed, like `blockingFunctions.triggers.beforeCreate.functionUri`, it will not actually pickup the changes.